core/templates/dev/head/pages/question_player/QuestionPlayerDirective.ts

```diff
@@ -34,14 +35,43 @@ oppia.directive('questionPlayer', [
         '$scope', '$rootScope', 'QuestionPlayerBackendApiService',
         function(
             $scope, $rootScope, QuestionPlayerBackendApiService) {
-          var questionPlayerConfig = $scope.getQuestionPlayerConfig();
-          QuestionPlayerBackendApiService.fetchQuestions(
-            questionPlayerConfig.skillList,
-            questionPlayerConfig.questionCount, true).then(function(result) {
-            $scope.currentQuestion = 1;
-            $scope.totalQuestions = result.length;
-            $scope.currentProgress = (
-              $scope.currentQuestion * 100 / $scope.totalQuestions);
+          $scope.questionPlayerConfig = $scope.getQuestionPlayerConfig();
+          $scope.currentQuestion = 0;
+          $scope.totalQuestions = 0;
+          $scope.currentProgress = 0;
+
+          var updateCurrentQuestion = function(currentQuestion) {
+            $scope.currentQuestion = currentQuestion;
+            updateQuestionProgression();
+          };
+
+          var updateTotalQuestions = function(totalQuestions) {
+            $scope.totalQuestions = totalQuestions;
+            updateQuestionProgression();
+          };
+
+          var updateQuestionProgression = function() {
+            if (getTotalQuestions() > 0) {
+              $scope.currentProgress = (
+                getCurrentQuestion() * 100 / getTotalQuestions());
+            } else {
+              $scope.currentProgress = 0;
+            }
+          };
+
+          var getCurrentQuestion = function() {
+            return $scope.currentQuestion;
+          };
+
+          var getTotalQuestions = function() {
```

Comment:
is there any specific reason for creating a function and not using the value directly?

Reply:
It was the practice in many other files such as ExplorationPlayerStateService/ContextService and so on. Also I personally prefer that just so that it's more semantic as opposed to $scope.<something>